selectors/admin_console.jsx

```diff
@@ -20,3 +23,35 @@ export const getAdminDefinition = createSelector(
 
 export const getAdminConsoleCustomComponents = (state, pluginId) =>
     state.plugins.adminConsoleCustomComponents[pluginId] || {};
+
+export const getConsoleAccess = createSelector(
+    getAdminDefinition,
+    getMySystemPermissions,
+    (adminDefinition, mySystemPermissions) => {
+        const consoleAccess = {read: {}, write: {}};
+
+        const has = (x) => mySystemPermissions.has(x);
+
+        const mapAccessValuesForKey = ([key]) => {
+            const permissionsForKey = ResourceToSysConsolePermissionsTable[key].filter((x) => mySystemPermissions.has(x));
+
+            consoleAccess.read[key] = permissionsForKey.length !== 0;
+            consoleAccess.write[key] = !consoleAccess.read[key] || !permissionsForKey.some((permission) => permission.startsWith('write'));
+
+            if (key === 'user_management') {
+                ['users', 'groups', 'teams', 'channels', 'permissions'].forEach((userManagementKey) => {
+                    const subKey = `${key}.${userManagementKey}`;
+
+                    const permissionsForSubkey = ResourceToSysConsolePermissionsTable[subKey].filter(has);
+
+                    consoleAccess.read[subKey] = permissionsForSubkey.length !== 0;
+                    consoleAccess.write[subKey] = !consoleAccess.read[key] || !permissionsForSubkey.some((permission) => permission.startsWith('write'));
+                });
+            }
+        };
+
+        Object.entries(adminDefinition).forEach(mapAccessValuesForKey);
+
+        return consoleAccess;
+    },
+);
```

Comment:
This can either go here in the webapp because it uses the admin definition, or it can live in the redux repo and accept the admin definition as a parameter.

Reply:
I think It makes most sense to keep it here